### DISCUSSION TITLE
How to connect with cloudflare

### DISCUSSION THREAD

**Discussion by simrskandou** [docker-mailserver/docker-mailserver#2765] [Q&A] [answered]
How to setup docker-mailserver connect with cloudflare?
do you have sample configuration?
thank you

**Comment by georglauterbach** [comment]
> How to setup docker-mailserver connect with cloudflare?

You don't. What do you even mean by this?

**Comment by polarathene** [comment, marked as answer]
If you look through the Cloudflare docs you'll find [this page](https://developers.cloudflare.com/dns/manage-dns-records/how-to/email-records/) (_the bottom section links to detailed pages on specific mail records_). There should also be some other doc pages where they tell you that their general proxy services aren't for mail-server DNS records.

Also be mindful about proxied IP [affecting your rDNS PTR record](https://developers.cloudflare.com/dns/additional-options/reverse-zones/). The first link I provided was only about an A record with an MX record for receiving mail, I don't think it's a good idea to have Cloudflare proxy your IP when sending mail as the IP from your server when sending outbound mail likely won't match the IP queried by DNS looking up your servers A record. [This other page of their docs](https://developers.cloudflare.com/dns/manage-dns-records/reference/proxied-dns-records/#ports-and-protocols) touches on this for SMTP, advising you not to proxy records involved in mail-servers, or use one of their other services.

We do not offer any support for such. I suggest you read the Cloudflare docs on the topic to get a better understanding.
